Rust/Rust.sublime-syntax

```diff
@@ -453,39 +499,47 @@ contexts:
     - include: block
 
   type:
-    - match: (?:{{support_type}}|{{identifier}})(?=<)
-      captures:
-        1: support.type.rust
+    # A low-level type. Typically you want type-any-identifier for the full
+    # type grammar.
+    - match: '{{identifier}}(?=<)'
       push: generic-angles
-    - match: \b(Self|i8|i16|i32|i64|i128|isize|u8|u16|u32|u64|u128|usize|f32|f64|bool|char|str)\b
+    - match: \b(Self|{{int_suffixes}}|{{float_suffixes}}|bool|char|str)\b
       scope: storage.type.rust
+    # stdsimd types
+    - match: |-
+        (?x)
+        \b(?:
+          [ium]8x(?:2|4|8|16|32) | [iu]8x64   |
+          [ium]16x(?:2|4|8|16)   | [iu]16x32  | # f16xN types might come later
+          [iumf]32x(?:2|4|8)     | [iuf]32x16 |
+          [iumf]64x(?:2|4)       | [iuf]64x8  |
+          m1x(?:64|32|16|8)                   | # 512-bit masks
+          __m(?:64|128|256)[di]? # __m512 should come later
+        )\b
+      scope: storage.type.rust
+    - match: '\bdyn\b(?!\s*::)(?=\s*(?:\(|{{lifetime}}|{{identifier}}))'
+      scope: storage.type.trait.rust
 
   generic-angles:
     - meta_scope: meta.generic.rust
-    - include: comments
-    - match: '->'
-      scope: punctuation.separator.generic.rust
-      push: generic-angles-contents
+    # -> to cover https://doc.rust-lang.org/std/boxed/trait.FnBox.html
+    - include: support-type
```

Comment:
As far as I can see `generic-angles` is pushed only by rules which contain a `(?=<)` lookahead. Hence the content of `-include: support-type` is never matched because `generic-angles-contents` is pushed immendiatelly and popped off by `(?=>)` only, which causes `generic-angles` to be popped off by line 257.

Removing that line does not break any syntax test provided with this package.

Reply:
Yea, I removed it. It was added in https://github.com/rust-lang/rust-enhanced/pull/173 and changed in https://github.com/rust-lang/rust-enhanced/pull/189, but it doesn't seem like it is needed anymore.